api/cloud_accounts_test.go

```diff
@@ -0,0 +1,284 @@
+//
+// Author:: Salim Afiune Maya (<afiune@lacework.net>)
+// Copyright:: Copyright 2021, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package api_test
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/lacework/go-sdk/api"
+	"github.com/lacework/go-sdk/internal/intgguid"
+	"github.com/lacework/go-sdk/internal/lacework"
+)
+
+func TestCloudAccountTypeAwsCtsSqs(t *testing.T) {
+	assert.Equal(t,
+		"AwsCtSqs", api.AwsCtSqsCloudAccount.String(),
+		"wrong cloud account type",
+	)
+}
+
+func TestFindCloudAccountType(t *testing.T) {
+	cloudFound, found := api.FindCloudAccountType("SOME_NON_EXISTING_INTEGRATION")
+	assert.False(t, found, "cloud account type should not be found")
+	assert.Equal(t, 0, int(cloudFound), "wrong cloud account type")
+	assert.Equal(t, "None", cloudFound.String(), "wrong cloud account type")
+
+	cloudFound, found = api.FindCloudAccountType("AwsCtSqs")
+	assert.True(t, found, "cloud account type should exist")
+	assert.Equal(t, "AwsCtSqs", cloudFound.String(), "wrong cloud account type")
+
+	//cloudFound, found = api.FindCloudAccountType("GcpCfg")
+	//assert.True(t, found, "cloud account type should exist")
+	//assert.Equal(t, "GcpCfg", cloudFound.String(), "wrong cloud account type")
+}
+
+func TestCloudAccountsGet(t *testing.T) {
+	var (
+		intgGUID    = intgguid.New()
+		vanillaType = "VANILLA"
+		apiPath     = fmt.Sprintf("CloudAccounts/%s", intgGUID)
+		vanillaInt  = singleVanillaCloudAccount(intgGUID, vanillaType, "")
+		fakeServer  = lacework.MockServer()
+	)
+	fakeServer.UseApiV2()
+	fakeServer.MockToken("TOKEN")
+	defer fakeServer.Close()
+
+	fakeServer.MockAPI(apiPath,
```

Comment:
not sure how your Mock API server works, but have a gander at this:
https://github.com/lacework/hawkeye/blob/d0f8dd4c241bf4734a54647bd8ee8ba26898fa97/usercallback/utils/http_test.go#L56

Might be useful to look at testing the potential flake-i-ness of our API endpoints with some 429s, 404s via ResponderFromMultipleResponses ...  ?

Reply:
MockAPI works similar to what you have coded in the link you provided, we mock specific APIs and test the result, the only difference is that, as a best practice, you should try to decouple tests and code by creating a different Go package, look at the file I have here, the package name is `api_test` which is how I expect consumers to access our Go `api` package.